### DISCUSSION TITLE
Some helpers for interdependent/mutually exclusive arguments

### DISCUSSION THREAD

**Discussion by wolfgang42** [bashly-framework/bashly#150] [Ideas]
My script has a couple of places where there are flags that are mutually exclusive, so I wrote a couple helper functions to make it easy to validate these cases, reproduced here in case someone else needs the same thing. (Public domain/CC0 if you're worried about licensing.)

Usage is something like:
```sh
args_mutually_exclusive --edit --no-edit # Cannot pass --edit and --no-edit simultaneously
args_exactly_one --create --import id # Must pass either --create, --import=file, or an existing id
```

(Obviously all of these `$args` entries must be declared and optional.)

```sh
args_mutually_exclusive() {
	declare -i count=0
	for arg in "$@"; do
		if [[ -v args["$arg"] ]]; then
			count=$count+1
		fi
	done
	if [[ "$count" -gt 1 ]]; then
		echo "No more than one of the following must be passed: $*" >&2
		exit 1
	fi
}

args_exactly_one() {
	declare -i count=0
	for arg in "$@"; do
		if [[ -v args["$arg"] ]]; then
			count=$count+1
		fi
	done
	if [[ "$count" != 1 ]]; then
		echo "Exactly one of the following must be passed: $*" >&2
		exit 1
	fi
}
```

**Comment by DannyBen** [comment]
Interesting. I wonder if there is any way to generalize these into validators (`bashly add validations`) or otherwise, straight into the config (without cognitive overload).

**Reply by DannyBen** [reply]
@wolfgang42 - did we solve this in one of our joint PRs? Or is it still something that needs a solution?

**Reply by wolfgang42** [reply]
This wasn't one of the things we covered in PRs, no. I didn't think it needed to be included in Bashly itself since it's easy to implement as external library functions, so I figured posting it as an idea was an easy way to share it.

**Reply by DannyBen** [reply]
I have added support for defining mutually exclusive flags.

**Comment by DannyBen** [comment]
Implementation is in #182
